differ/differ.go

```diff
@@ -232,6 +345,62 @@ func NewFilesDifferWithFDPool(file1, file2 string, fdPool *fdp.FdPool, collectio
 	return differ, nil
 }
 
+func constructHlv(docCas uint64, importCAS uint64, bucketUUID string, hlvbytes []byte) (*hlv.HLV, error) {
+	cvCas, cvSrc, cvVer, pvMap, mvMap, err := crMeta.ParseHlvFields(docCas, hlvbytes)
+	if err != nil {
+		return nil, err
+	}
+	//need to convert the uuid(hex) to base64 i.e. hlv required format
+	bucketId, err1 := hlv.UUIDtoDocumentSource(bucketUUID)
+	if err1 != nil {
+		return nil, err1
+	}
+	var Hlv *hlv.HLV
+	var err2 error
+	if docCas == importCAS {
+		Hlv, err2 = hlv.NewHLV(bucketId, cvCas, cvCas, cvSrc, cvVer, pvMap, mvMap)
+	} else {
+		Hlv, err2 = hlv.NewHLV(bucketId, docCas, cvCas, cvSrc, cvVer, pvMap, mvMap)
+	}
+	if err2 != nil {
+		return nil, err2
+	}
+	return Hlv, nil
+}
+
+func extractHlvAndImportCas(XattrIterator *xdcrBase.XattrIterator, bucketUUID string, docCas uint64) (hlv *hlv.HLV, importCas uint64, err error) {
```

Comment:
Is there a reason why you tie HLV and ImportCAS together in this function?

Reply:
No, but I remember seeing it done that way on the XDCR side, so I preferred to follow it.